x-pack/test/functional/apps/rollup_job/hybrid_index_pattern.js

```diff
@@ -0,0 +1,107 @@
+    const indexName = 'rollup-to-be';
+    const now = new Date();
+    const pastDates = [
+      datemath.parse('now-1d', { forceNow: now }),
+      datemath.parse('now-2d', { forceNow: now }),
+      datemath.parse('now-3d', { forceNow: now }),
+    ];
+
+    it('create hybrid index pattern', async () => {
+      //Create data for rollup job to recognize.
+      //Index past data to be used in the test.
+      await pastDates.map(async (day) => {
+        await es.index(mockIndices(day));
+      });
+
+      await retry.waitForWithTimeout('waiting for 3 records to be loaded into elasticsearch.', 10000, async () => {
+        const response = await es.indices.get({
+          index: 'to-be*',
+          allow_no_indices: false
+        });
+        return Object.keys(response).length === 3;
+      });
+
+      await retry.try(async () => {
+        //Create a rollup for kibana to recognize
+        await es.transport.request({
+          path: `/_rollup/job/${rollupJobName}`,
+          method: 'PUT',
+          body: {
+            'index_pattern': 'to-be*',
+            'rollup_index': indexName,
+            'cron': '*/10 * * * * ?',
+            'groups': {
+              'date_histogram': {
+                'fixed_interval': '1000ms',
+                'field': '@timestamp',
+                'time_zone': 'UTC'
+              }
+            },
+            'timeout': '20s',
+            'page_size': 1000
+          }
+        });
+      });
+
+
+      await pastDates.map(async (day) => {
+        await es.index(mockRolledUpData(rollupJobName, day));
+      });
+
+      //Add data for today, 1,2 and 3 days into the future.
+      const futureDates = [
+        datemath.parse('now', { forceNow: now }),
+        datemath.parse('now+1d', { forceNow: now }),
+        datemath.parse('now+2d', { forceNow: now }),
+        datemath.parse('now+3d', { forceNow: now }),
+      ];
+
+      //Index live data to be used in the test.
+      await futureDates.map(async (day) => {
+        await es.index(mockIndices(day, 'live-data'));
+      });
+
+      await PageObjects.common.navigateToApp('settings');
+      await PageObjects.settings.createIndexPattern('live*,' + indexName, '@timestamp', false);
+
+      await PageObjects.common.navigateToApp('discover');
+      const hits = await PageObjects.discover.getHitCount();
+
+      //Number of hits should be at least 7. Due to CI being slow, the legacy data may not be there.
+      //3 for rolled up data, 3 for legacy data that was rolled up, and 4 for the live data.
+      expect(Number.parseInt(hits)).to.equal(7);
+    });
+
+    after(async () => {
+      // Delete the rollup job.
+      es.transport.request({
+        path: `/_rollup/job/${rollupJobName}`,
+        method: 'DELETE',
+      });
+      await es.indices.delete({ index: 'rollup*' });
+      await es.indices.delete({ index: 'live*' });
+      await es.indices.delete({ index: 'live*,rollup-to-be*' });
```

Comment:
Do we need to delete `live*` again here? And wouldn't deleting `rollup*` already capture `rollup-to-be*`?

Reply:
Fixed.